StreamAPI/Main.java

```diff
@@ -0,0 +1,214 @@
+
+        System.out.println(ibanWithStars(persons));
+
+        System.out.println("------------10,11-------------");
+        Person p6 = new Person("Anna Mons", 16, address1 = null);
+        Person p7 = new Person("Koenigsegg", 35, address2 = null);
+        Person p8 = new Person("Peter the Great", 47, address3 = null);
+        Person p9 = new Person("John Smith", 16, address3 = null);
+        Person p10 = new Person("Ivanov", 47, address4 = null);
+        List<Person> list2 = Arrays.asList(p6, p7, p8, p9, p10);
+        System.out.println("Total age of those older than 17 is " + ageAll(list2));
+        System.out.println(legalAge(list2));
+
+    }
+
+    //Task1 addresses of those older than 17
+    public static List<Address> addressMoreSeventeen(List<Person> input) {
+        List<Address> addrMoreSeventeen = input
+                .stream()
+                .filter(p -> p.getAge() > 17)
+                .map(Person::getAddress)
+                .collect(Collectors.toList());
+        return addrMoreSeventeen;
+
+    }
+
+    //2. List of names without duplicates
+    public static Set<String> withoutDuplicates() {
+        Stream<String> streamOfNames = Stream.of("Ivan", "Petr", "Ivan", "Semen", "Petr");
+        Set<String> withoutDupl = streamOfNames.collect(Collectors.toSet());
+        return withoutDupl;
+    }
+
+    //3. return map, where age is a key and list of persons is a value
+    public static Map<Integer, List<Person>> mapGetAge(List<Person> input) {
+        Map<Integer, List<Person>> map = input
+                .stream()
+                .collect(Collectors.groupingBy(Person::getAge));
+
+        return map;
+
+    }
+    //4.return map with Person as a key  and list of his/her accounts as value
+
+    public static Map<Person1, List<BankAccount>> mapPersonAsKeyAccountAsValue(List<BankAccount> input) {
+        Map<Person1, List<BankAccount>> mapPersonAsKey = input
+                .stream()
+                .collect(Collectors.groupingBy(BankAccount::getOwner));
+        return mapPersonAsKey;
+
+    }
+
+    //5.IBANNs list with * after third symbol.
+    public static List<String> getListofIbannsWithStars(List<BankAccount> input) {
+        List<String> listOfIbannsWithStars = input
+                .stream()
+                .map(a -> replaceWithStars(a.getIBANN()))
+                .collect(Collectors.toList());
+        return listOfIbannsWithStars;
+
+
+    }
+
+    public static String replaceWithStars(String input) {
+        char[] chars = input.toCharArray();
+        for (int i = 3; i < input.length(); i++)
+            chars[i] = '*';
+        return new String(chars);
+    }
+
+
+    //6. method accepts a sentence and
+    // returns number of words starting with a certain letter
+    public static long numberOfWords(String input, String w) {
+        long count = Stream.of(input.split(" "))
+                .filter(s -> s.startsWith("a"))
```

Comment:
I think it returns words, which start with 'a', not with `w`.

Reply:
Exactly. Should be w there instead of  an "a".